extensions/omni_rest/src/postgrest_rpc.sql

```diff
@@ -0,0 +1,132 @@
+create or replace function _postgrest_function_call_arguments (fn regproc, passed_arguments text[])
+    returns text immutable
+    language sql
+    as $$
+    select
+        string_agg(format('%1$I => $%2$s :: %3$I', name, idx, t.typname), ', ' order by idx)
+    from
+        pg_proc p,
+        unnest(proargnames, proargtypes, proargmodes)
+    with ordinality as _ (name, type, mode, idx)
+    join pg_type t on t.oid = type
+where
+    type is not null
+    and p.oid = fn
+    and name::text = any (passed_arguments)
+$$;
+
+create or replace function _postgrest_function_ordered_argument_values (fn regproc, passed_arguments text[], passed_values jsonb)
+    returns jsonb immutable
+    language sql
+    as $$
+    select
+        jsonb_agg(passed_values ->> (name::text)
+        order by idx)
+    from
+        pg_proc p,
+        unnest(proargnames, proargtypes, proargmodes)
+    with ordinality as _ (name, type, mode, idx)
+    join pg_type t on t.oid = type
+where
+    type is not null
+    and p.oid = fn
+    and name::text = any (passed_arguments)
+$$;
+
+create procedure postgrest_rpc (request omni_httpd.http_request, outcome inout omni_httpd.http_outcome, settings postgrest_settings default postgrest_settings ())
+language plpgsql
+as $$
+declare
+    function_reference regproc;
+    query text;
+    namespace text;
+    result jsonb;
+    arguments_definition text;
+    argument_values jsonb;
+    passed_arguments text[];
+begin
+    if outcome is distinct from null then
+        return;
+    end if;
+    if request.method in ('GET', 'POST') then
+        -- FIXME: write guard clause checking for function
+        if split_part(request.path, '/', 2) <> 'rpc' then
```

Comment:
Note (not a change request): the path should eventually go into `postgrest_settings` (with the default set to `/rpc`).